src/material/chips/chip.ts

```diff
@@ -232,6 +240,7 @@ export class MatChip extends _MatChipMixinBase implements FocusableOption, OnDes
 
   constructor(public _elementRef: ElementRef<HTMLElement>,
               private _ngZone: NgZone,
+              @Inject(DOCUMENT) private _document: any,
```

Comment:
Since you're only using the `_document` inside the constructor you don't have to declare it as a private property.

Also adding a new required constructor parameter is a breaking change. It should be added as an optional parameter initially with a `@breaking-change` annotation so that we can eventually make it required. See how it's done for `_changeDetectorRef` below.

Reply:
What should the `@breaking-change` annotation version be?

Given it's now optional, if it doesn't exist, should we fallback to just not having ripples? If not, what's the fallback?

Reply:
The `@breaking-change` should just say which version we're supposed to clean it up. E.g. `@breaking-change 11.0.0 Make _document a required parameter`.

It defeats the purpose a little bit, but while it's optional we have to fall back to just reading the global `document`.

Reply:
Falling back to global `document` sounds good. Sorry, what's the specific version number we need to clean it up in? How can I find that number?

Reply:
We usually do next major version + 1 for breaking changes.